### PR TITLE
CLN: de-duplicate paths in tslibs

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/_libs/tslibs/period.pyx [hunk 1]
```diff
@@ -964,23 +964,8 @@ def dt64arr_to_periodarr(const int64_t[:] dtarr, int freq, tz=None):
     """
     cdef:
         int64_t[:] out
-        Py_ssize_t i, l
```
Comment (jreback) [inline pandas/_libs/tslibs/period.pyx:967]: do we even need this routine any longer? (e.g. just call localize_dt64arr_to_period in the caller)

File: pandas/_libs/tslibs/resolution.pyx [hunk 2]
```diff
@@ -28,27 +28,19 @@ cdef:
 
 # ----------------------------------------------------------------------
 
-cpdef resolution(const int64_t[:] stamps, tz=None):
+def resolution(const int64_t[:] stamps, tz=None):
```
Comment (jreback) [inline pandas/_libs/tslibs/resolution.pyx:31]: not called in cython anylonger?

Comment (jbrockmendel) [reply pandas/_libs/tslibs/resolution.pyx:31]: not sure if it ever was

File: pandas/_libs/tslibs/period.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/period.pyx:967]: sure

**Pull request merged**